cliconfig/credentials/store.go

```diff
@@ -0,0 +1,10 @@
+package credentials
+
+import "github.com/docker/engine-api/types"
+
+// Store is an interface that credentials providers must implement
+type Store interface {
+	Add(types.AuthConfig) error
+	Get(serverURL string) (string, string, error)
```

Comment:
yep, it only uses the server address for now, it could be extended to use serverAddress && username.